distributed/client.py

```diff
@@ -1383,7 +1383,12 @@ def wait_for_workers(
             raise ValueError(
                 f"`n_workers` must be a positive integer. Instead got {n_workers}."
             )
-        return self.sync(self._wait_for_workers, n_workers, timeout=timeout)
+        try:
+            return self.cluster.wait_for_workers(n_workers, timeout)
+        except AttributeError:
+            # Most likely, either self.cluster is None, or the cluster has not
+            # implemented a wait_for_workers method
```

Comment:
nit: You added wait_for_workers to the base class, i.e. every cluster will come equipped with this. I typically prefer dealing with these situations by being explicit, i.e. `if cluster is None; ...`

Reply:
I agree, changed it to an if-else